src/impl/transact_log_handler.cpp

```diff
@@ -657,20 +657,23 @@ void advance_with_notifications(BindingContext* context, SharedGroup& sg, Func&&
     if (observers.empty() && (!notifiers || notifiers.version())) {
         notifiers.before_advance();
         func(TransactLogValidator());
-        auto new_version = sg.get_version_of_current_transaction();
+        auto new_version = sg->get_version_of_current_transaction();
         if (context && old_version != new_version)
             context->did_change({}, {});
+        // did_change() could close the Realm. Just return if it does.
+        if (!sg)
```

Comment:
Yes, destroying the `NotifierPackage` without delivering it should be fine and not delivering the notifications seems like the only sensible thing to do.